x-pack/plugins/lens/public/pie_visualization/pie_visualization.tsx

```diff
@@ -0,0 +1,236 @@
+  id: 'lnsPie',
+
+  visualizationTypes: [
+    {
+      id: 'donut',
+      largeIcon: CHART_NAMES.donut.icon,
+      label: CHART_NAMES.donut.label,
+    },
+    {
+      id: 'pie',
+      largeIcon: CHART_NAMES.pie.icon,
+      label: CHART_NAMES.pie.label,
+    },
+  ],
+
+  getLayerIds(state) {
+    return state.layers.map(l => l.layerId);
+  },
+
+  clearLayer(state) {
+    return {
+      shape: state.shape,
+      layers: state.layers.map(l => newLayerState(l.layerId)),
+    };
+  },
+
+  getDescription(state) {
+    if (state.shape === 'donut') {
+      return CHART_NAMES.donut;
+    }
+    return CHART_NAMES.pie;
+  },
+
+  switchVisualizationType: (visualizationTypeId, state) => ({
+    ...state,
+    shape: visualizationTypeId as PieVisualizationState['shape'],
+  }),
+
+  initialize(frame, state) {
+    return (
+      state || {
+        shape: 'donut',
+        layers: [newLayerState(frame.addNewLayer())],
+      }
+    );
+  },
+
+  getPersistableState: state => state,
+
+  getSuggestions({
+    table,
+    state,
+    keptLayerIds,
+  }: SuggestionRequest<PieVisualizationState>): Array<
+    VisualizationSuggestion<PieVisualizationState>
+  > {
+    if (
+      keptLayerIds.length > 1 ||
+      (keptLayerIds.length && table.layerId !== keptLayerIds[0]) ||
+      (state && table.changeType === 'unchanged') ||
+      table.columns.some(col => col.operation.dataType === 'date')
+    ) {
+      return [];
+    }
+
+    const [slices, metrics] = partition(table.columns, col => col.operation.isBucketed);
+
+    if (slices.length === 0 || metrics.length > 1) {
+      return [];
+    }
+
+    const title =
+      table.changeType === 'unchanged'
+        ? i18n.translate('xpack.lens.pie.suggestionLabel', {
+            defaultMessage: 'As {chartName}',
+            values: { chartName: state ? CHART_NAMES[state.shape].label : CHART_NAMES.donut.label },
+          })
+        : i18n.translate('xpack.lens.pie.suggestionOf', {
+            defaultMessage: '{chartName} {operations}',
+            values: {
+              chartName: state ? CHART_NAMES[state.shape].label : CHART_NAMES.donut.label,
+              operations:
+                table.label ||
+                table.columns
+                  .map(col => col.operation.label)
+                  .join(
+                    i18n.translate('xpack.lens.datatable.conjunctionSign', {
+                      defaultMessage: ' & ',
+                      description:
+                        'A character that can be used for conjunction of multiple enumarated items. Make sure to include spaces around it if needed.',
+                    })
+                  ),
+            },
+          });
+
+    return [
+      {
+        title,
+        score: 0.6,
+        state: {
+          shape: state ? state.shape : 'donut',
+          layers: [
+            {
+              layerId: table.layerId,
+              slices: slices.map(col => col.columnId),
+              metric: metrics[0].columnId,
```

Comment:
This causes a bug when you delete metric, here's the reproducing video:

https://drive.google.com/file/d/15xKAxtNy4lXZ_fuTkRtE8pH2UOYw7DeB/view?usp=sharing

Reply:
Another one to reproduce this one is to try to add one more slices level to treemap chart (when there's already 2 existing) by dropping the field to the visualization. Eg. try to drop `email` here:
![image](https://user-images.githubusercontent.com/4283304/79754117-4e568180-8317-11ea-9b9d-2ea6ceb65068.png)